src/main/java/io/chazza/advancementapi/AdvancementAPI.java

```diff
@@ -238,14 +252,13 @@ public String toString() {
         }
     }
 
-
     //BEGIN CLASSES
     @Builder(builderMethodName = "hiddenbuilder")
     public static class Condition {
         protected String name;
-        protected JSONObject set;
+        protected JsonObject set;
```

Comment:
That's where I based my prediction in. I noticed you should have the ability to make it either a string or a json object since different advancements take different properties. 
The Minecraft:bred animal needs a json object while the Minecraft:brew potion needs a string. 

If you are aware of a way to make a json object behave as a string when adding into other json objects let me know.